### PR TITLE
Fix Mailgun header format for Reply-To

### PR DESCRIPTION
**Pull request by loduis**
In this moment is sent in lower case so: h:reply-to
You can see the article is h:Reply-To
https://help.mailgun.com/hc/en-us/articles/4401814149147-Adding-A-Reply-To-Address

| Q             | A
| ------------- | ---
| Branch?       | 5.4 for features / 4.4, 5.4 or 6.0 for bug fixes 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by fabpot** [comment]
Headers are not case sensitive. Are you sure it makes a difference?

**Comment by loduis** [comment]
Some mail clients are reply to our from address **no-reply** after upgrading to **symfony mailer** and when I check my old mail sent with **mailgun-php** the only thing different that I find is that, but I would have to review in more depth the behavior of mailgum-php and the handling of the reply-to header

Before:
image
After:
image

File: src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php [hunk 1]
```diff
@@ -139,7 +139,7 @@ private function getPayload(Email $email, Envelope $envelope): array
             if (\in_array($prefix, ['h:', 't:', 'o:', 'v:']) || \in_array($name, ['recipient-variables', 'template', 'amp-html'])) {
                 $headerName = $name;
             } else {
-                $headerName = 'h:'.$name;
+                $headerName = 'h:'.mb_convert_case($name, MB_CASE_TITLE, 'UTF-8');
```
Comment (OskarStark) [inline src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php:142]:
```suggestion
                $headerName = 'h:'.tarlala_convert_case($name, MB_CASE_TITLE, 'UTF-8');
```

**Comment by nicolas-grekas** [comment]
Closing in favor of #45816
Thanks for letting us know about the issue!

**Pull request closed without merging**